beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -2070,3 +2070,35 @@ func (s *Server) GetGenesis(w http.ResponseWriter, r *http.Request) {
 	}
 	httputil.WriteJson(w, resp)
 }
+
+// GetDepositSnapshot returns the deposit snapshot.
+func (s *Server) GetDepositSnapshot(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetDepositSnapshot")
+	defer span.End()
+
+	eth1data, err := s.BeaconDB.ExecutionChainData(ctx)
+	if err != nil {
+		httputil.HandleError(w, "Could not retrieve execution chain data: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	snapshot := eth1data.DepositSnapshot
+	if snapshot == nil {
+		httputil.HandleError(w, "No Finalized Snapshot Available", http.StatusNotFound)
```

Comment:
based on the function below maybe a check for len(snapshot.Finalized)==0 should be added here too instead of just an empty snapshot